### PR TITLE
Add local `keys` operation

### LINKED ISSUE #7: Local pipeline operations adapter
## Context

**PipelineOperations** interface provides interface with generic  pipeline operations and it allows us to encapsulate a pipeline framework (Apache Beam or Apache Spark or other).

 One of the options is to run “locally”, without any pipeline framework. For example it might be convenient to perform DP aggregations on Pandas DataFrames (it's outside of scope of this issue).

## Goal

 Implement **LocalOperations** class, which is inherited from **PipelineOperations** and implements all abstract methods using pure Python.

Eg. 

```
def map(self, col, fn, stage_name: str):
	return map(col, fn)
```

where col is iterable.

**Note**: stage_name is needed only for Beam, and it might be ignored for **LocalPipelineOperations**. 

**Note**: because of nuances of how DP pipelines work, all operations should be lazy, luckily Python has a concept of [generators](https://wiki.python.org/moin/Generators), which should simplify lazy implementation. 

**Note**: it is better to split an implementation of different methods in different pull requests.

## Links
TODO: add a link to PipelineOperations sources when [pull request](https://github.com/OpenMined/PipelineDP/pull/2) is submitted

### PR DESCRIPTION
**Pull request by zachferr**
## Description
Add local `keys` operation. This closes  #7 
Sorry for the delay here, I went on vacation then I had to catch up on my main work. 

## Affected Dependencies
N/A

## How has this been tested?
- Added a unit test

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Review by lovroprepolec — CHANGES_REQUESTED** [review]

File: pipeline_dp/pipeline_operations.py [hunk 1]
```diff
@@ -145,8 +145,8 @@ def has_public_partition_key(pk_val):
             VALUES: col,
             IS_PUBLIC: public_partitions
         } | "Aggregating elements by values and is_public partition flag " >>
-                beam.CoGroupByKey() | "Filterding data from public partitions"
-                >> beam.ParDo(PartitionsFilterJoin()))
+            beam.CoGroupByKey() | "Filterding data from public partitions"
```
[the remark this answers is not in the document]
Comment (zachferr) [reply pipeline_dp/pipeline_operations.py:148]: The change was accidental I didn't even realize my formatter did it. When you say maintain do you mean change it back to how it was or keep it as is?

[the remark this answers is not in the document]
Comment (lovroprepolec) [reply pipeline_dp/pipeline_operations.py:148]: No worries. Please change it back. Thank you!

File: tests/pipeline_operations_test.py [hunk 2]
```diff
@@ -206,6 +206,13 @@ def test_local_filter_by_key_remove(self):
                                         "Public partition filtering")
         self.assertEqual(result, [(7, (1, 7, 1)), (9, (3, 9, 1))])
 
+    def test_local_keys(self):
+        self.assertEqual(list(self.ops.keys([])), [])
+
+        example_list = [(1, 2), (2, 3), (3, 4), (4, 8)]
+
+        self.assertEqual(list(self.ops.keys(example_list)), [1, 2, 3, 4])
+
```
[the remark this answers is not in the document]
Comment (zachferr) [reply tests/pipeline_operations_test.py:215]: Done

File: pipeline_dp/pipeline_operations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (zachferr) [reply pipeline_dp/pipeline_operations.py]: Fixed the typo and reverted the change.

**Pull request merged**